core/src/test/java/com/sequenceiq/cloudbreak/controller/v4/StackV4ControllerTest.java

```diff
@@ -67,4 +75,10 @@ void generateImageCatalogInternalTest() {
 
         verify(stackOperations).generateImageCatalog(NameOrCrn.ofName(STACK_NAME), WORKSPACE_ID);
     }
+
+    @Test
+    public void testCcmUpgrade() {
+        when(stackCcmUpgradeService.upgradeCcm(any(), any())).thenReturn(new FlowIdentifier(FlowType.FLOW, "1"));
+        ThreadBasedUserCrnProvider.doAs(USER_CRN, () -> underTest.upgradeCcmByNameInternal(STACK_NAME, USER_CRN));
```

Comment:
Note: Impersonating the user and executing as `USER_CRN` is indeed carried out in the real implementation, but is not strictly necessary for the UT that uses stubbing for `stackCcmUpgradeService.upgradeCcm()`. :smile: No need to change, just a note.